lib/rules/sort-comp.js

```diff
@@ -87,6 +87,7 @@ module.exports = {
       groups: {
         lifecycle: [
           'displayName',
+          'props',
```

Comment:
This approach is recommended by the [Flow documentation](https://flowtype.org/docs/react.html). I agree that outside of Flow you would never ever want to do this, and that’s why I think it’s fine to make this addition. If you’re ever assigning to `props` (even in Flow you aren’t assigning), it will result in a bug. However, when you do whether it be for TypeScript, Flow, or any other arbitrary type checker or JavaScript intelligence tool there is an idiomatic place to put the `props` attribute, the top of the component class.

I’m also going to cc @thejameskyle on this one to see if he wants to make a comment.